src/arns.lua

```diff
@@ -387,9 +387,14 @@ local function getTotalArNSNamesBetweenTimestampsUnsafe(startTimestamp, endTimes
 		if record.type == "permabuy" then
 			totalActiveNames = totalActiveNames + 1
 		elseif record.type == "lease" and record.endTimestamp then
+			-- record is active between the start and end timestamps
 			if record.startTimestamp <= startTimestamp and record.endTimestamp >= endTimestamp then
```

Comment:
Why is the start timestamp relevant here? It's not clear what this function's criteria are for names being between the timestamp. I would think that it's active if:
- record.endTimestamp is greater than startTimestamp of interest (e.g. it didn't expire before the startTimestamp of interest)
- record.endTimestamp is also less than or equal to endTimestamp of interest (still not expired)

record.startTimestamps seem kind of irrelevant depending on what's intended here.

Reply:
if you wanted to slice of records that were active in a window - you have to account for when they started. this eliminates records that start after the provided current timestamp as they do not exist for the entirety of the boundaries

Reply:
i updated this to drop the boundary constraint an only concern itself with epoch start timestamp. this reduces some of the logic you asked about [560dda9](https://github.com/ar-io/ar-io-network-process/pull/352/commits/560dda98224d770dfeaacec77ea1fc9ebedb5395)